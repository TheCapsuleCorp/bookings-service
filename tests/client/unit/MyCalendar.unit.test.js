import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import MyCalendar from '../../../client/src/components/MyCalendar.jsx';
import bookingsSample from '../../../database/dummy_data.js';

configure({ adapter: new Adapter() });

describe('MyCalendar', () => {
  it('should render correctly', () => {
    const output = shallow(
      <MyCalendar listingId={bookingsSample.star_rating}></MyCalendar>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

describe('MyCalendar', () => {
  it('should render fully and correctly', () => {
    const output = shallow(
      <MyCalendar listing={bookingsSample}></MyCalendar>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});