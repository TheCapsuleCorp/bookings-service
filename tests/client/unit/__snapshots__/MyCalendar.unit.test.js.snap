// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCalendar should render correctly 1`] = `
<div
  className="calendar-container"
>
  <ClickOutComponent
    onClickOut={[Function]}
  >
    <div>
      <button
        className="calendar-btn align-right"
        onClick={[Function]}
      >
        Check In
      </button>
      <button
        className="calendar-btn align-left"
        onClick={[Function]}
      >
        Check Out
      </button>
    </div>
    <div
      className="modal"
      id="myModal"
      style={
        Object {
          "border": "1px grey solid",
        }
      }
    >
      <Calendar
        calendarType="US"
        maxDetail="month"
        minDetail="century"
        onClickDay={[Function]}
        prev2ButtonDisabled={true}
        returnValue="range"
        selectRange={false}
        showNavigation={true}
        showNeighboringMonth={false}
        tileClassName={[Function]}
        tileDisabled={[Function]}
        value={
          Array [
            2018-03-09T01:43:00.348Z,
            null,
          ]
        }
        view="month"
      />
      <div
        className="white-background caption"
      >
        Minimum stay varies
      </div>
      <div
        className="white-background caption"
      >
        Updated today
      </div>
      <div />
    </div>
  </ClickOutComponent>
</div>
`;

exports[`MyCalendar should render fully and correctly 1`] = `
<div
  className="calendar-container"
>
  <ClickOutComponent
    onClickOut={[Function]}
  >
    <div>
      <button
        className="calendar-btn align-right"
        onClick={[Function]}
      >
        Check In
      </button>
      <button
        className="calendar-btn align-left"
        onClick={[Function]}
      >
        Check Out
      </button>
    </div>
    <div
      className="modal"
      id="myModal"
      style={
        Object {
          "border": "1px grey solid",
        }
      }
    >
      <Calendar
        calendarType="US"
        maxDetail="month"
        minDetail="century"
        onClickDay={[Function]}
        prev2ButtonDisabled={true}
        returnValue="range"
        selectRange={false}
        showNavigation={true}
        showNeighboringMonth={false}
        tileClassName={[Function]}
        tileDisabled={[Function]}
        value={
          Array [
            2018-03-09T01:43:00.359Z,
            null,
          ]
        }
        view="month"
      />
      <div
        className="white-background caption"
      >
        Minimum stay varies
      </div>
      <div
        className="white-background caption"
      >
        Updated today
      </div>
      <div />
    </div>
  </ClickOutComponent>
</div>
`;
